packages/es-dev-server/src/utils/resolve-module-imports.js

```diff
@@ -184,18 +184,17 @@ async function resolveModuleImportsWithConfig(importer, source, cfg) {
       lastIndex = end;
     } else if (dynamicImportIndex >= 0) {
       // dynamic import
-      const dynamicStart = start + 1;
-      const dynamicEnd = end - 1;
-
+      const literals = [`\``, "'", '"'];
+      const stringSymbol = literals.find(char => source.substring(start, end).includes(char));
+      const dynamicStart = source.indexOf(stringSymbol, start) + 1;
+      const dynamicEnd = source.indexOf(stringSymbol, dynamicStart);
       const [importee, importeeSuffix] = getImportee(source.substring(dynamicStart, dynamicEnd));
-      const stringSymbol = source[dynamicStart - 1];
-      const isStringLiteral = [`\``, "'", '"'].includes(stringSymbol);
+      const isStringLiteral = literals.includes(stringSymbol);
```

Comment:
`stringSymbol` is taken from the list of literals, so there is no need to check if it's in the list again. probably the clearest would be to check if `dynamicStart !== -1`